libgnucash/engine/Account.cpp

```diff
@@ -2481,73 +2477,122 @@ xaccAccountSetDescription (Account *acc, const char *str)
-        g_value_set_static_string (&v, value);
-        qof_instance_set_path_kvp (QOF_INSTANCE (acc), &v, path);
-        g_value_unset (&v);
-    }
-    else
-    {
-         qof_instance_set_path_kvp (QOF_INSTANCE (acc), nullptr, path);
-    }
-    mark_account (acc);
+    qof_instance_set_path_kvp<const char*> (QOF_INSTANCE(acc), val, path);
     xaccAccountCommitEdit(acc);
 }
 
+static const char*
+get_kvp_string_path (const Account *acc, const Path& path)
+{
+    auto rv{qof_instance_get_path_kvp<const char*> (QOF_INSTANCE(acc), path)};
+    return rv ? *rv : nullptr;
+}
+
 static void
-set_kvp_string_tag (Account *acc, const char *tag, const char *value)
+set_kvp_account_path (Account* acc, const Path& path, const Account* kvp_account)
 {
-    set_kvp_string_path (acc, {tag}, value);
+    std::optional<GncGUID*> val;
+    if (kvp_account)
+        val = guid_copy(xaccAccountGetGUID (kvp_account));
+
+    xaccAccountBeginEdit(acc);
+    qof_instance_set_path_kvp<GncGUID*> (QOF_INSTANCE(acc), val, path);
+    xaccAccountCommitEdit(acc);
 }
 
-static const char*
-get_kvp_string_path (const Account *acc, std::vector<std::string> const & path,
-                     GValue *v)
+static Account*
+get_kvp_account_path (const Account *acc, const Path& path)
 {
-    *v = G_VALUE_INIT;
-    if (acc == nullptr) return nullptr; // how to check path is valid??
-    qof_instance_get_path_kvp (QOF_INSTANCE (acc), v, path);
-    return G_VALUE_HOLDS_STRING (v) ? g_value_get_string (v) : nullptr;
+    auto val{qof_instance_get_path_kvp<GncGUID*> (QOF_INSTANCE(acc), path)};
+    return val ? xaccAccountLookup (*val, gnc_account_get_book (acc)) : nullptr;
 }
 
-static const char*
-get_kvp_string_tag (const Account *acc, const char *tag, GValue *v)
+static void
+set_kvp_boolean_path (Account *acc, const Path& path, gboolean option)
+{
+    set_kvp_string_path (acc, path, option ? "true" : nullptr);
+}
+
+static gboolean
+get_kvp_boolean_path (const Account *acc, const Path& path)
+{
+    auto slot{QOF_INSTANCE(acc)->kvp_data->get_slot(path)};
+    if (!slot) return false;
+    switch (slot->get_type())
+    {
+    case KvpValueImpl::Type::INT64:
+        return slot->get<int64_t>() != 0;
+    case KvpValueImpl::Type::STRING:
+        return g_strcmp0 (slot->get<const char*>(), "true") == 0;
+    default:
+        return false;
+    }
+}
+
+static void
+set_kvp_int64_path (Account *acc, const Path& path, std::optional<gint64> value)
 {
-    return get_kvp_string_path (acc, {tag}, v);
+    xaccAccountBeginEdit(acc);
+    qof_instance_set_path_kvp<int64_t> (QOF_INSTANCE(acc), value, path);
+    xaccAccountCommitEdit(acc);
+}
+
+static const std::optional<gint64>
+get_kvp_int64_path (const Account *acc, const Path& path)
+{
+    return qof_instance_get_path_kvp<int64_t> (QOF_INSTANCE(acc), path);
 }
 
 void
 xaccAccountSetColor (Account *acc, const char *str)
 {
-    set_kvp_string_tag (acc, "color", str);
+    g_return_if_fail(GNC_IS_ACCOUNT(acc));
```

Comment:
No need for this (or any of the other's you've added) once `qof_instance_set_path_kvp` and `qof_instance_get_path_kvp` do the checks. Mind the additional checks are harmless and do report the error a little closer to its source so arguably it makes for a better trace file.